src/osp/Resource/AssetImporter.cpp

```diff
@@ -269,6 +280,17 @@ DependRes<Texture2D> osp::AssetImporter::compile_tex(
     return package.add<Texture2D>(imageData.name(), std::move(tex));
 }
 
+DependRes<Magnum::GL::Texture2D> AssetImporter::compile_tex(
```

Comment:
i wonder if it wouldn't be more ergonomic to have these "compile_XYZ" functions instead be template functions "compile" with the internal logic changing based on the type of the package.
```
template<typename PACKAGE_T>
void AssetImporter::compile(std::string_view imageDataName, Package& package) = delete;
template<>
void AssetImporter::compile<ImageData2D>(std::string_view imageDataName, Package& package)
{
    auto imgData = package.get<ImageData2D>(imageDataName);
    if (imgData.empty())
    {
        return { };
    }
    // do the same treatment for the function taking a DependRes<PACKAGE_T>
    return compile(imgData, package);
}
```

Reply:
Hmm, I'll consider it. I don't think we need any more versions beyond meshes and textures. Assuming these would be the only two variants needed, is there any point still to do this?

Reply:
If only two versions are needed then it doesn't really matter either way.